core/session/src/lib.rs

```diff
@@ -29,6 +29,8 @@ client::decl_runtime_apis! {
 	/// Session keys runtime api.
 	pub trait SessionKeys {
 		/// Generate a set of session keys with optionally using the given seed.
+		/// The keys should be stored within the keystore exposed via runtime
```

Comment:
It is correct that we do it this way, but this is not a requirement of this api.

Reply:
This documentation makes it a requirement.